pyflagser/flagser.py

```diff
@@ -144,7 +145,8 @@ def flagser_weighted(adjacency_matrix, max_edge_length=None, min_dimension=0,
         If ``True``, computes persistent homology for the directed flad complex
         determined by `adjacency_matrix`. If ``False``, computes persistent
         homology for the undirected flag complex obtained by considering all
-        edges as undirected.
+        edges as undirected, and it is therefore sufficient (but not necessary)
```

Comment:
In the weighted case, I'm less sure: couldn't there be conflicting weights? What is the behaviour then?

Reply:
This behavior is undocumented in C++ flagserso I had to read the code. What happens is that if you have the following edges and weights:

5 8 2.3
8 5 1.2

Only a single edge is added: 5 8 with weight 2.3 because 8 > 5.

Reply:
> Only a single edge is added: 5 8 with weight 2.3 because 8 > 5

Just to check: this is more of a random implementational decision than a sound mathematical reason, right?

Reply:
Yes, definitely.

Reply:
Great, thanks for putting the work into checking it @gtauzin. Thinking about it more the rule 8 > 5 is precisely the fact that one of the two is upper diagonal, and the other one is not. So it looks like maybe the rule could be sound: C++ flagser only reads the upper diagonal in the undirected case. Agree?

Reply:
Btw also, what you just said makes me worried whether this is also what happes in the unweighted case, in which case the current explanation would be misleading/incorrect.

Reply:
@ulupo No, flagser reads both the upper and the lower diagonal but if both entry (i j) and (j i) are present with different filtration values, it will only consider the filtration value of (i j) for the undirected edge i - j

Reply:
Hmm, I'm still confused, sorry :(

> it will only consider the filtration value of (i j) for the undirected edge i - j

Why (i j) instead of (j i)?

Reply:
Yes and it is an arbitrary choice.

Reply:
Great, thanks for the clarification. Still, as this is a little more delicate than in the unweighted case, I feel we can try to give a bit more guidance on the expected input. A suggestion could be along these lines:
"If ``False``, computes persistent homology for the undirected filtered flag complex obtained by considering all weighted edges as undirected. More precisely, the weight associated to the undirected edge between node i and node j, with i < j, is: a) the (i, j) entry of `adjacency_matrix`, if explicitly set; b) the (j, i) entry of `adjacency_matrix`, if explicitly set and if entry (i, j) was not explicitly set; c) ``np.inf``, if neither enties were explicitly set. When ``False``, it is therefore sufficient (but not necessary) to pass an upper-triangular matrix."

Reply:
I would write something a bit simpler

"If ``False``, computes persistent homology for the undirected flag complex obtained by considering all edges as undirected, and it is therefore sufficient (but not necessary) to pass an upper-triangular matrix. When ``False``, if two directed edges corresponding to the same undirected edge are assigned different weights, only the one on the upper triangular part of the adjacency matrix is considered."

Reply:
@gtauzin thanks! Yes it does read much better that way. I'd still prefer to be explicit about "undirected filterered flag complex" (here and also in the case `True` above) and even "all weighted edges".